## QProMS - Quantitative PROteomics Made Simple

QProMS is a shiny app that enables easy but powerful and reproducible analyses for label-free proteomics data. It works out of the box with  major data-dependent and data-independent search engine results (MaxQuant, FragPipe, Spectronaut, DIA-NN, AlphaPept) as well as custom result tables. It can produce publication-quality figures and export HTML reports and parameter files for sharing and reproducing results. It can handle multiple simultaneous comparisons between different experimental conditions.

### Installation or running online

Open the app [here](https://bioserver.ieo.it/shiny/app/qproms). The app can also be run locally. Open Rstudio (make sure you have RTools installed along with R>4.0) and open this project from a local download or by loading it from version control. Then type:


    renv::restore()


To install all dependencies. You can then load the app by


    shiny::runApp()

    
and heading to the address in your browser.


### Getting started

The app guides you through a typical analysis workflow for proteomics:

1. Data upload and experimental design annotation
2. Quality control and handling missing values
3. Statistical analysis (clustering, volcano and more)
4. Network and functional analysis
5. Exporting results.

In all sections, the options pre-selected are defaults that should apply to most scenarios including global proteome profiling, AP-MS and proximity labelling MS.

#### Upload page

Upload search engine results. Files should have a column for gene IDs, and columns for quantitation of each protein in the respective conditions. Several search engines are supported by default. File names are arbitrary, but the files to upload for each search engine are


| Software    | File to upload (default name)  | Comment                                                                                                                                                            | 
|-------------|--------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------| 
| MaxQuant    | ProteinGroups.txt              | Must contain "Gene Names" column. QProMS includes automatic parsing of reverse, contaminants and only identified by site and handling of ambiguous protein groups. |
| FragPipe    | combined_protein.tsv           | QProMS has automatic annotation of contaminants included by Philosopher/FragPipe.  |
| Spectronaut |                                |  | 
| Dia-NN      | report.unique_genes_matrix.txt | |  
| AlphaPept   | results_proteins.csv           | |  

Select the organism of your data and and click on "upload". If your organism is not available or gene names are missing, the app will still work, but some functions such as GO enrichment will be disabld. The app will automatically detect which result table you have if automatic detection fails because of a non-standard table. You may then select which columns of your table represent gene names and which ones are intensities from the left side menu.

If you have worked with QProMS before and are trying to reproduce an analysis or an earlier session, you can tick on "load parameters" and upload the QProMS .yml file generated by "Generate report" (see below) together with your protein table.

All intensities are automatically log2 transformed in the upload phase unless specified otherwise. 

After clicking on "upload" you can edit your experimental design in the table that appears below. It is modelled after the MaxQuant and FragPipe experimental design tables.

| Column name | Meaning                                                                                                                                                                   |
|-------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| key         | Unique name for the column that is only used by QProMS (not editable)                                                                                                     | 
| condition   | The experimental condition (e.g. control/tumor, time point etc). Can be letters and/or numbers. All replicates of one condition must have the same condition name         |
| keep        | Whether to keep or discard this column for the QProMS analysis                                                                                                            |

Once you are done editing the experimental design, click "Apply". The green "Start" button should appear. Clicking it will unlock the rest of the app.

#### Preprocessing


The "filter data" page displays protein counts, upset plot, intensity distributions and the distribution of coefficients of variation (CVs). 

The left panel for "parameters" can be expanded to reveal some settings that can be changed. All settings will propagate through later steps of the analysis and saved in the report.

The user can tweak filtering based by valid values based on percentage valid values in at least one experimmental condition (group), in all groups, or in total.

Normalization across samples using variance stabilizing normalization (VSN) can be toggled on here.

Other options available here are dependent on the search engine used in the upload page: 

- For MaxQuant, the app automatically filters proteins to those with minimum 2 peptides detected (editable by the user), and removes contaminants, reverse and only identified by site (also editable by the user).

- For FragPipe, contaminants are automatically removed based on the [CrapOme list](https://reprint-apms.org/?q=about) utilized by FragPipe and Philosopher during database construction.



In the missing data page, the user can analyse the distribution of missing values and perform imputation. The plots highlight the intensity distributions and the effects of imputation. The contribution from imputation can be highlighted in the "Effect of imputation: after" plot. The distribution can be broken down by sample by clicking on the bar at the bottom of the page.

In the blue ribbon for "parameters" the algorithm for imputation may be selected. There are 2 modes: mixed imputation, introduced in the [QProMS publication](XXX), and the Perseus algorithm, adapted from [here](http://www.coxdocs.org/doku.php?id=perseus:user:activities:matrixprocessing:imputation:replacemissingfromgaussian). In the Perseus algorithm, missing values are replaced by sampling from a down-shifted gaussian distribution.

The mean of the imputed distribution can be positioned relative to the mean experimental intensity in terms of standard deviations with the "down shift" parameter. The width of the imputed distribution can be adjusted by the "scale" parameter.

Mixed imputation recognises two types of missing data: missing (MAR) at random and missing not at random (MNAR). Missing at random values (i.e. those missing from a single replica in a condition) are imputed with the mean of the value of that protein in the other replicas. Missing not at random (completely missing from a particular experimental condition) are imputed with the Perseus-style algorithm.

In the last panel, the table post data filtering and imputation (if imputation is used) is available for export and interactive searching.


#### Principal component analysis (PCA)

The page displays a principal component analysis in 2d and 3d. The top of the page highlights the variance explained by each principal component. These plots are useful as quality control to ensure that replicas cluster with each other and separate across different experimental conditions (i.e. that most of the variance is contributed by changing of experimental condition).

#### Correlation

Correlation matrix for the data. Scatter plots dynamically update by clicking on different squares in the correlation matrix. Any number of scatterplots may be generated by selecting different experiments.

Users can select the type of correlation analysis by clicking on the parameters tab to set Pearson (default), Spearmann or Kendall correlation.


#### Rank




#### Volcano

This page guides the user through differential analysis and generation of the volcano plot. 

In Inputs, the user selects which conditions to compare. Each comparison is written as "enriched_vs_control". Multiple comparisons may be selected, each of which will be displayed as its own volcano plot.

In "Parameters", the user selects the type of statistical test for generation of the p-value and the FDR correction (truncation).

- Welch's T-test (default): two-sample t-test that is more reliable when populations have unequal variances and different sample sizes.
- Student's T-test: two-sample t-test that assumes populations have equal variance.
- limma (linear models for microarray and RNA-Seq): Bayesian univariate test relying on row-wise linear models to estimate likelihood of differental expression. More information [here](https://academic.oup.com/nar/article/43/7/e47/2414268).

The "paired" option is off by default but can be toggled if a paired test is to be performed. This is desirable if the comparison is between two samples that are correlated.

"Fold change" is the minimum fold change (in log2 units) to consider as significant.

Alpha is the level of significance after correction for multiple testing (Truncation) is applied. Several corrections are available, with the default being Benjamini/Hochberg FDR correction.

